### PR TITLE
Removing redundant code

### PR DESCRIPTION
**Pull request by CjHare**
## PR description
`ECIESEncryptionEngine` was inspired by the BouncyCastle implementation of `IESEngine`. 
The Bouncy Castle code used `V` (which is redundant in our code), with it being initialised to an empty array, however in `processBlock()` `V` could be set to another value, but as that method was not carried across to Besu, which means in Besu it would always be an empty array.

Although `ECIESEncryptionEngine` has no unit test, the logic is covered by these unit tests:
- `EncryptedMessageTest` 
- `ECIESHandshakeTest`

## Fixed Issue(s)

Remedies two LGTM warnings.
Removes dead code.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -276,8 +276,8 @@ task checkMavenCoordianteCollisions {
         def coordinate = it.publishing?.publications[0].coordinates
         if (coordinates.containsKey(coordinate)) {
           throw new GradleException("Duplicate maven coordinates detected, ${coordinate} is used by " +
-              "both ${coordinates[coordinate]} and ${it.path}.\n" +
```
Comment (CjHare) [inline build.gradle]: `applySpotless` made these changes, do we want these? ...is my spotless broken?

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/handshake/ecies/ECIESEncryptionEngine.java [hunk 2]
```diff
@@ -209,22 +206,16 @@ public BytesValue encrypt(final BytesValue in, final byte[] macData)
...
     }
 
     mac.doFinal(T, 0);
 
-    // Output the triple (V,C,T).
-    final byte[] Output = new byte[V.length + len + T.length];
-    System.arraycopy(V, 0, Output, 0, V.length);
-    System.arraycopy(C, 0, Output, V.length, len);
-    System.arraycopy(T, 0, Output, V.length + len, T.length);
+    // Output the triple (C,T).
```
Comment (pinges) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/handshake/ecies/ECIESEncryptionEngine.java]: This is not a triple anymore!

Comment (CjHare) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/handshake/ecies/ECIESEncryptionEngine.java]: Well spotted 👍

File: build.gradle (hunk 1 above)
Comment (shemnon) [reply build.gradle]: Yes.  Because it's going to keep making them.

Comment (CjHare) [reply build.gradle]: Well, consistency is king 👍

**Pull request merged**